ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -599,6 +599,12 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 		}
 	}
 
+	// if RayCluster is suspended, delete all pods and skip reconcile
+	if instance.Spec.Suspend != nil && *instance.Spec.Suspend {
+		filterLabels = client.MatchingLabels{common.RayClusterLabelKey: instance.Name}
+		return r.DeleteAllOf(ctx, &corev1.Pod{}, client.InNamespace(instance.Namespace), filterLabels)
```

Comment:
Would emitting an event be useful, to track suspension events? Or maybe a condition would be better suited?